src/redux/selectors/cellSets/getCellSetsHierarchyByKeys.js

```diff
@@ -2,9 +2,13 @@ import createMemoizedSelector from 'redux/selectors/createMemoizedSelector';
 
 import getCellSetsHierarchy from './getCellSetsHierarchy';
 
-const getCellSetsHierarchyByKeys = (keys) => (hierarchy) => (
-  keys.map((key) => hierarchy.find((child) => child.key === key))
-);
+const getCellSetsHierarchyByKeys = (keys) => (hierarchy) => {
+  if (hierarchy.length === 0) return [];
```

Comment:
Would it be nicer if we have a wrapper selector around all, which checks if the cellSets defined (check cellSets.accessible for example)?

Reply:
That's already being done by one of its input selectors (`getCellSetsHierarchy`)

Reply:
If I don't have that, this selector will return an array of undefined's which was the bug this if is addressing.

Reply:
I meant to have this check (hierarchy.length === 0) in the wrapper, not sure if there are cases where 
hierarchy.length === 0 is alright when we get the cellsets right?

Reply:
What I meant is that that check it is already being done in the input selector for this one if you check the file and its input selector `{ inputSelectors: getCellSetsHierarchy() }`.

I could change this check for a cellSets.accessible? I am not sure if a new selector would make sense here

Reply:
Let me know what you think

Reply:
Alternatively I could just remove this check that returns `[]` and instead just handle this in the components that use the selector by checking whether they are undefined or not or something

Reply:
let me know what you think